data/doc_ock.py

```diff
@@ -25,22 +27,25 @@ def event_name(cls) -> str:
 
 
 def post_embedding_creation(user_id: str, config_string: str) -> Any:
-    return _post_event(user_id, config_string, enums.EmbeddingState.INITIALIZING.value)
+    return __post_event_threaded(user_id, config_string, enums.EmbeddingState.INITIALIZING.value)
 
 
 def post_embedding_encoding(user_id: str, config_string: str) -> Any:
-    return _post_event(user_id, config_string, enums.EmbeddingState.ENCODING.value)
+    return __post_event_threaded(user_id, config_string, enums.EmbeddingState.ENCODING.value)
 
 
 def post_embedding_finished(user_id: str, config_string: str) -> Any:
-    return _post_event(user_id, config_string, enums.EmbeddingState.FINISHED.value)
+    return __post_event_threaded(user_id, config_string, enums.EmbeddingState.FINISHED.value)
 
 
 def post_embedding_failed(user_id: str, config_string: str) -> Any:
-    return _post_event(user_id, config_string, enums.EmbeddingState.FAILED.value)
+    return __post_event_threaded(user_id, config_string, enums.EmbeddingState.FAILED.value)
+
 
+def __post_event_threaded(user_id: str, config_string: str, state: str) -> Any:
+    daemon.run(__post_event, user_id, config_string, state)
 
-def _post_event(user_id: str, config_string: str, state: str) -> Any:
+def __post_event(user_id: str, config_string: str, state: str) -> Any:
```

Comment:
the config string is filled with the model. No all have a model (cohere) maybe we need to change it to model + platform & then combine both for the field for config_string

Reply:
you are right, but do we wanna change this now?

Reply:
done